core/pkg/server/stream_init.go

```diff
@@ -147,6 +168,15 @@ func NewFileTransferManager(
 		fileTransferStats,
 	)
 
+	// Set proxy
+	fileTransferRetryClient.HTTPClient.Transport.(*http.Transport).Proxy = ProxyFn(settings.Proto.GetXProxies().GetValue())
```

Comment:
similar to comment in `api.go`---we should create a Transport rather than type-casting